packages/react-native/Libraries/AppDelegate/RCTRootViewFactory.mm

```diff
@@ -95,8 +95,21 @@ @interface RCTRootViewFactory () <RCTCxxBridgeDelegate> {
 @end
 
 @implementation RCTRootViewFactory {
-  RCTRootViewFactoryConfiguration *_configuration;
   __weak id<RCTTurboModuleManagerDelegate> _turboModuleManagerDelegate;
+  __weak id<RCTHostDelegate> _hostDelegate;
+  RCTRootViewFactoryConfiguration *_configuration;
+}
+
+- (instancetype)initWithTurboModuleDelegate:(id<RCTTurboModuleManagerDelegate>)turboModuleManagerDelegate hostDelegate:(id<RCTHostDelegate>)hostdelegate configuration:(RCTRootViewFactoryConfiguration*)configuration {
+  if (self = [super init]) {
+    _configuration = configuration;
+    _hostDelegate = hostdelegate;
+    _contextContainer = std::make_shared<const facebook::react::ContextContainer>();
+    _reactNativeConfig = std::make_shared<const facebook::react::EmptyReactNativeConfig>();
+    _contextContainer->insert("ReactNativeConfig", _reactNativeConfig);
```

Comment:
Can the initialization of these fields be moved to a common init? Now its duplicated with `initWithConfiguration: andTurboModuleManagerDelegate`

Reply:
Yes, I agree. The other initializer should call this one.

```objc
// Deprecate this.
- (instancetype)initWithConfiguration:(RCTRootViewFactoryConfiguration *)configuration
        andTurboModuleManagerDelegate:(id<RCTTurboModuleManagerDelegate>)turboModuleManagerDelegate
{
  // Usually, the TMMD is implemented by AppDelegate which also conforms to the  HostDelegate
  return [self initWithTurboModuleDelegate:turboModuleManagerDelegate hostDelegate:turboModuleManagerDelegate configuration:(RCTRootViewFactoryConfiguration*)configuration
}
```

Reply:
we can add a check:
```objc
id<RCTHostDelegate> hostDelegate = [turboModuleManagerDelegate conformsToProtocol:@protocol(RCTHostDelegate)] ? (id<RCTHostDelegate>)turboModuleManagerDelegate : nil
  return [self initWithTurboModuleDelegate:turboModuleManagerDelegate hostDelegate:turboModuleManagerDelegate configuration:(RCTRootViewFactoryConfiguration*)configuration
```
This should work fine and we won't lose any functionality.

Reply:
Done 👍